test/libsolidity/semanticTests/inlineAssembly/external_function_pointer_selector.sol

```diff
@@ -0,0 +1,22 @@
+contract C {
+	function testFunction() external {}
+
+	function testYul() public returns (uint32) {
+		function() external fp = this.testFunction;
+		uint selectorValue = 0;
+
+		assembly {
+			selectorValue := fp.selector
+		}
+
+		return uint32(bytes4(bytes32(selectorValue)));
+	}
+	function testSol() public returns (uint32) {
+		return uint32(this.testFunction.selector);
+	}
```

Comment:
If `.selector` in Solidity is `uint32` (right-aligned in a 32-byte slot) why do we want it left-aligned in Yul? Wouldn't it be more consistent to have it right-aligned in both cases?

Reply:
it is not, it is a `bytes4`.